### ISSUE TITLE
Language: Bangla, Monkeytype cannot recognize characters ড়, য় 

### ISSUE THREAD

**Issue by Oymate** [monkeytypegame/monkeytype#4468]
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes, but sometimes it didn't

### Account name

sajsajrob

### Account config

{"theme":"serika_dark","themeLight":"serika","themeDark":"serika_dark","autoSwitchTheme":false,"customTheme":false,"customThemeColors":["#323437","#e2b714","#e2b714","#646669","#2c2e31","#d1d0c5","#ca4754","#7e2a33","#ca4754","#7e2a33"],"favThemes":[],"showKeyTips":true,"showLiveWpm":false,"showTimerProgress":true,"smoothCaret":"medium","quickRestart":"off","punctuation":false,"numbers":false,"words":50,"time":30,"mode":"time","quoteLength":[1],"language":"bangla","fontSize":1.5,"freedomMode":false,"difficulty":"normal","blindMode":false,"quickEnd":false,"caretStyle":"default","paceCaretStyle":"default","flipTestColors":false,"layout":"default","funbox":"none","confidenceMode":"off","indicateTypos":"off","timerStyle":"mini","colorfulMode":false,"randomTheme":"off","timerColor":"main","timerOpacity":"1","stopOnError":"off","showAllLines":false,"keymapMode":"off","keymapStyle":"staggered","keymapLegendStyle":"lowercase","keymapLayout":"overrideSync","keymapShowTopRow":"layout","fontFamily":"Roboto_Mono","smoothLineScroll":false,"alwaysShowDecimalPlaces":false,"alwaysShowWordsHistory":false,"singleListCommandLine":"manual","capsLockWarning":true,"playSoundOnError":false,"playSoundOnClick":"off","soundVolume":"0.5","startGraphsAtZero":true,"showOutOfFocusWarning":true,"paceCaret":"off","paceCaretCustomSpeed":100,"repeatedPace":true,"pageWidth":"125","accountChart":["on","on","on"],"minWpm":"off","minWpmCustomSpeed":100,"highlightMode":"letter","alwaysShowCPM":false,"ads":"result","hideExtraLetters":false,"strictSpace":false,"minAcc":"off","minAccCustom":90,"showLiveAcc":false,"showLiveBurst":false,"monkey":false,"repeatQuotes":"off","oppositeShiftMode":"off","customBackground":"","customBackgroundSize":"cover","customBackgroundFilter":[0,1,1,1,1],"customLayoutfluid":"qwerty#dvorak#colemak","monkeyPowerLevel":"off","minBurst":"off","minBurstCustomSpeed":100,"burstHeatmap":false,"britishEnglish":false,"lazyMode":false,"showAverage":"off","tapeMode":"off"}

### Current Behavior

Monkeytype cannot recognize the letters ড়, য় (maybe even ঢ় though I was not able to get any words that include ঢ়). I believe this happens because it uses some old encoding scheme where ড+় is used, where the correct letter would be ড়.

### Expected Behavior

Monkeytype recognizes ড়, ঢ়, য়

### Steps To Reproduce

1. Go to Monkeytype
2. Switch language to Bangla
3. Restart words until you get words such as নয়, উড়ন্ত, পড়া, শয়ন, গাঢ় etc.
4. Try to type these words using Linux i-bus keyboard or Windows Avro phonetic/Jatiyo keyboard.
5. Monkeytype can't recognize characters

### Environment

- OS:Win 10, PopOS latest
- Browser:Firefox
- Browser Version:115.0.2

**Comment by Miodec** [comment]
How do you achieve those characters? Which keys do I press?

**Comment by Oymate** [comment]
If you use Avro, then you can change layout by pressing F12(don't press Shift+Alt messes up keyboard on windows). Then from dropdown you can select phonetic or Jatiyo. If you select jatiyo then press P/Shift+E/Shift+P, to get these letters. On phonetic you have to type phonetically in english. Like নয় is 'noy' followed by space/enter to select dropdown text.

On Linux install Bangla(Bangladesh) Jatiyo Layout and with Windows key+Space switch keyboard and type P/Shift+E/Shift+P.

Or the simplest solution for you, copy paste from unicode block https://symbl.cc/en/unicode/blocks/bengali/#Additional_consonants in additional consonant section.

**Comment by Miodec** [comment]
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/e089865f-f97a-402d-8774-1c01bc9e426d)
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/a598ae28-99e7-450e-8f27-e139c74668c2)
![image](https://github.com/monkeytypegame/monkeytype/assets/13181393/a7497cfb-4662-4018-90f6-077bfb9e2d22)

Which one is best? Which keys do i press then? Which ones are broken?

**Comment by Oymate** [comment]
Can you send screenshot of the first two option in 'Shift' mode? Thanks

**Comment by Oymate** [comment]
BTW, which app are you using? I might be able to give clearer response if I know.

**Comment by Miodec** [comment]
Thats just the MacOS layout selector

**Comment by Oymate** [comment]
Ok, the second layout is MS InScript layout. In order to get ড়,ঢ়,য়
- Press Alt+Opening Square bracket/third bracket(ড়)
- Press Alt+Shift+Opening Square bracket/third bracket(ঢ়)
- Press Shift+?(য়)

**Comment by Oymate** [comment]
I would like to remind you that কিিি is intended rendering while  ক ি ি ি ি ি is not. The reason ড় gets split might be because in previous(like 90's) version of unicode there was no ড়, ড and the fota was combined to get the letter. Subsequent application brought in the character ড়।
When you do implement changes related to this case please give me screenshots as example so I can assure you it is correct.

**Comment by Miodec** [comment]
I understand the separate rendering is not intended. I also said that there is no way around it without completely reworking the website and straight up removing one of the main features (whch is per key highlight)

**Comment by Oymate** [comment]
Actually the per key highlight works as intended and expected for other words and characters.

**Comment by Miodec** [comment]
It doesn't. When you go to zen mode and type anything, no characters get connected.
<img width="1444" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/3223bcee-1146-409d-af8d-8c4e174db7bc">

**Comment by Oymate** [comment]
Wait is zen deployed or in testing?

**Comment by Miodec** [comment]
Zen is live

**Comment by Oymate** [comment]
Interesting, it works in normal mode but not zen mode. In normal mode characters after the main letter are highlighted properly while characters before the main letter are not. Is it possible to implement [this](https://www.latex-project.org/publications/2020-UFi-TUB-tb127fischer-bangla.pdf), page 70 glyph colouring?

**Comment by Oymate** [comment]
If it's not possible to fix these without breaking changes, just implement proper input normalization for ড়,ঢ় and য় , I will forgo Zen mode no problem.

**Comment by Miodec** [comment]
Nevermind, i was wrong. These characters are displayed correctly if you just switch the language to bangla.. 
<img width="958" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/bfc3c363-fa37-4e00-98a6-edd36f3e55a6">

**Comment by Miodec** [comment]
Alright, well... going full circle, how do i type উড়ন্ত ? Which keys do i press? Because so far i can type ড় and it gets recognised

**Comment by Miodec** [comment]
<img width="701" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/be8d6a3d-0c3e-41c9-b76f-de0577d31238">

Looks to be working correctly for me

**Comment by Miodec** [comment]
Yeah i did. Interesting, let me play around.

**Comment by mm-crj** [comment]
I have the same problem with য় . I couldn't get how you guys fixed it.
![image](https://github.com/monkeytypegame/monkeytype/assets/14097405/91bd08f2-f1e7-47a1-b0fe-a2d1a0f25724)

**Comment by Miodec** [comment]
I think i found a solution, ill deploy it to the live site later today - please test and let me know.

**Comment by mm-crj** [comment]
I tested it for the letter 'য়' that works but I will check the other characters like it and let you know. However there  is another issue that I would like to draw your attention to here.

**Comment by mm-crj** [comment]
I think it is kind of working, but it detects something after the letter after য়. So each time I have to back space before typing forwar.

**Comment by Miodec** [comment]
How do you type that problematic letter?

**Comment by mm-crj** [comment]
I am on ubuntu, I use the Baisakhi layout
![image](https://github.com/monkeytypegame/monkeytype/assets/14097405/9e79af80-dd1d-4f57-8e40-b0d4c3e625bc)
then its shift+y.

**Comment by Miodec** [comment]
Seems like im getting a different character when pressing shift+y. Which keys is on this layout?

<img width="356" alt="image" src="https://github.com/monkeytypegame/monkeytype/assets/13181393/51d2d926-5aa0-4bec-930d-1976fb0b7f09">

**Comment by Miodec** [comment]
য় this one, right?

**Comment by Miodec** [comment]
So, I cant seem to reproduce the issue with this character.

Does it only happen in time mode?
Does it happen when this character appears in the first 200 words of the test? Or only after the first 200?
Does it happen in custom mode?

**Comment by Oymate** [comment]
I just tested again and it's working perfectly! Thanks for fixing it. 
(P.S. there's spelling mistake for a word which should be [কিন্তু ](https://www.english-bangla.com/bntoen/index/%E0%A6%95%E0%A6%BF%E0%A6%A8%E0%A7%8D%E0%A6%A4%E0%A7%81)but is written in the test as কিন্ত, )
